scripts/benchmarks/merge-benchmark-data.js

```diff
@@ -0,0 +1,43 @@
+const BENCHMARK_LAYERS = [
+  '../../staging/benchmarks/benchmarks_base_2017.json',
+  '../../staging/benchmarks/benchmarks_updates_20171109.json',
+  '../../staging/benchmarks/benchmarks_readmission_20171109.json'
+];
+
+const UNIQUE_COLUMN_CONSTRAINT = [
+  'measureId',
+  'benchmarkYear',
+  'performanceYear',
+  'submissionMethod'
+];
+
+const getBenchmarkKey = (benchmark) => {
+  let benchmarkKey = '';
+  UNIQUE_COLUMN_CONSTRAINT.forEach((keyName) => {
+    benchmarkKey += `${benchmark[keyName]}|`;
+  });
+
+  return benchmarkKey;
+};
+
+const mergeBenchmarkLayers = (benchmarkLayers) => {
+  const mergedBenchmarks = new Map();
+
+  benchmarkLayers.forEach((benchmarkLayer) => {
+    const benchmarkFile = require(benchmarkLayer);
+    benchmarkFile.forEach((benchmark) => {
+      mergedBenchmarks.set(getBenchmarkKey(benchmark), benchmark);
```

Comment:
is full replacement of previous benchmarks OK? or are there any fields included in previous benchmark files that we want to keep despite new info in subsequent layers?

Reply:
Full replacement seems like the best way forward - for maintainability and given the way we receive updates is via updated CSVs (i.e. it would be easy to miss an update if we were only looking in certain columns).

Reply:
Cool. Double checking, what im currently doing is a full replacement? I still need to overall key on a set of keys.

Reply:
Yeah this is a 'full replacement' because the entire benchmark value is getting replaced (not merging via Object.assign).